### DISCUSSION TITLE
Adding @Since javadoc information to public APIs of graphql-java library

### DISCUSSION THREAD

**Discussion by nutharsh** [graphql-java/graphql-java#2235] [General]
Hey All, when I was going through the public APIs of graphql-java library, I found that sometimes,it is not easy to find from when a particular public API got introduced.

For example, the below class annotated with @PublicApi makes it part of public API of the library.

```
/**
 * This helps you build {@link graphql.GraphQLError}s and also has a quick way to make a  {@link graphql.execution.DataFetcherResult}s
 * from that error.
 */
@PublicApi
public class GraphqlErrorBuilder {

    private String message;
    private List<Object> path;
    private List<SourceLocation> locations = new ArrayList<>();
    private ErrorClassification errorType = ErrorType.DataFetchingException;
    private Map<String, Object> extensions = null;
    
```

In my opinion, it is important part of the documentation to add `@Since` annotation to Javadoc from the library standpoint (true in this case) and as it is intended for someone else to use.

**Comment by bbakerman** [comment]
I agree it would be very nice to have. 

 But as an volunteer open source project, world class javadoc is not something we have invested in.  Ensuring release management down to the java doc level is not something that I think we can pull off in effort terms.
